### ISSUE TITLE
Questions about this convolutional autoencoder

### ISSUE THREAD

**Issue by planaria158** [fchollet/keras#2841]
Hi All,

I've been experimenting with autoencoders for reconstructing input images.  Had good success with MNIST digits.  Now wanting to try CIFAR10 images.

At the end of this message is the network I've coded up. 

In this network below, I've experimented with:
- number of filters (64, 32, 16, 8)
- weight initialization for the Conv layers.
- activation functions in the decoding layers.
- overall depth of the network.
- Sometimes I get nan's for loss.  More recently, this example below returns negative values for loss and has a great deal of trouble converging to anything (accuracies are only about 0.003! after 100 epochs and loss is large negative value).

Some other details:
- Training set is only 100 images
- Running on my Macbook pro (El Capitan)
- Theano backend
- Using cpu (not gpu)
- Keras 1.0.3
- Theano 0.8.2
- .theanorc: [global] floatX=float64  (I've tried both float32 and float64)  device=cpu  optimizer=fast_run  (fast_compile tends to create NANs)  mode=FAST_RUN  exception_verbosity=high

Here is my fitting and the first 2 epochs:

``` sh
autoencoder.fit(x_train, x_train, nb_epoch=2,  batch_size=20, 
          shuffle=True, verbose=1, validation_data=(x_test, x_test))

Train on 100 samples, validate on 20 samples
Epoch 1/2
100/100 - 0s - loss: -10.1022 - acc: 0.0515 - val_loss: -49.2346 - val_acc: 0.0464
Epoch 2/2
100/100- 0s - loss: -67.9034 - acc: 0.0470 - val_loss: -111.7369 - val_acc: 0.0464
```

It basically moves very little from what you see above (even after a few hundred epochs).  Also, the negative loss values (based on log loss) trouble me since I believe the log loss value should always be > 0.

Visualizing what the network predicts shows blank images. 

Here is the complete iPython notebook of my network:
[CIFAR10.ipynb.zip](https://github.com/fchollet/keras/files/288093/CIFAR10.ipynb.zip)

**Comment by joelthchao** [comment]
- Input is not normalized
```python
x_train = x_train.astype('float32') / 255.
x_test = x_test.astype('float32') / 255.
```
- Activation

`relu` for decoder is ok for me. It's not necessary to do `sigmoid` in the intermediate layers.

- Traing size

Larger training set and batch size will be better for learning.

**Comment by planaria158** [comment]
Thanks Joel!  Works great now.

re Training size: I was using a very small set just to make sure the model was well behaved before scaling up.

**Comment by manallllll** [comment]
hi, 
please, I have a problem with shallow denoising autoencoder, the result is "nan". I check the dataset, it doesn't have "nan" values. I tried different activation function, it is still the same problem 

this is my code 
model %>%
  layer_dense(units=15,activation="relu", input_shape=c(26))%>%
  layer_dense(units=26)
model%>%compile(loss='mean_squared_error',optimizer="adam")

history<-model%>%fit(noisy.data, original.data,
                     epochs = 20, 
                     batch_size = 100,
                     validation_split=0.1 )
and this is the result 
Epoch 1/20
87317/87317 [==============================] - 4s 40us/step - loss: nan - val_loss: nan
anyone help me, please??
